web/app/scripts/src/services/proportion-field.es.js

```diff
@@ -0,0 +1,62 @@
+/*
+ *
+ * This module calculates the survey completion percentage/
+ *
+ */
+
+import { angular } from '../libraries';
+
+export const moduleProperties = {
+    moduleName: 'incuna-surveys.proportion-field',
+    componentName: 'proportionField'
+};
+
+const module = angular.module(moduleProperties.moduleName, []);
+
+module.service(moduleProperties.componentName, [
+    function () {
+
+        this.buildFields = function (choices, defaults, autoId) {
+            return choices.map((choice, index) => {
+                return Object.assign(
+                    {},
+                    defaults,
+                    {
+                        label: choice,
+                        id: `${autoId}-${index}`
+                    }
+                );
+            });
+        };
+
+        this.calculateTotal = function (values) {
+            return Object.keys(values).reduce((total, key) => {
+                if (!values[key]) {
+                    return total;
+                }
+                let value = values[key] && parseInt(values[key], 10);
+                if (Number.isNaN(value)) {
+                    return total;
+                }
+                return total + value;
```

Comment:
Since `addPercentages` is using falseyness to determine `NaN`, why not also do that here instead of using the explicit `Number.isNaN()`?

So
```js
if (Number.isNaN(value)) {
    return total;
}
return total + value;
```
could become
```js
return (total + value) || total;
```
because `total + value`, if `NaN`, would be falsey and it would move on to `return total`.